wagtail/contrib/simple_translation/views.py

```diff
@@ -47,20 +48,18 @@ def post(self, request, **kwargs):  # pragma: no mccabe
         form = self.get_form()
 
         if form.is_valid():
+            include_subtree = form.cleaned_data["include_subtree"]
+
             with transaction.atomic():
                 for locale in form.cleaned_data["locales"]:
                     if isinstance(self.object, Page):
-                        self.object.copy_for_translation(locale)
-                        if form.cleaned_data["include_subtree"]:
-
-                            def _walk(current_page):
-                                for child_page in current_page.get_children():
-                                    child_page.copy_for_translation(locale)
-
-                                    if child_page.numchild:
-                                        _walk(child_page)
+                        action = CopyForTranslationAction(
+                            page=self.object,
+                            locale=locale,
+                            include_subtree=include_subtree,
```

Comment:
Could we add ``user=request.user`` here? This is a problem with the current implementation so might be worth updating the tests too.

Reply:
Sure!